### PR TITLE
EBNT-256 Tests for both tf versions, all client implementations and respective fixes

### PR DESCRIPTION
**Pull request by i-a-andrianov**
In this PR I did:

1. modified tox to test tensorflow v1 after tensorflow v2 (reinstallation of library is performed for that)

1. improved tests for local Ebonite client and applied them for in-memory, local and remote (PostgreSQL+Minio) clients

1. made `id` of `EboniteObject` subclasses an integer instead of string: local repo used UUIDs while SQL used auto-incremented integers, this leads to hard to understand method signatures (`NameOrIdOrObject`), to inconsistency in behaviour and to bugs (SQLite unlike PostgreSQL has very weak checking of queries)

1. fixed wrong indexes and queries in SQL repo (again, SQLite backend unlike PostgreSQL has very weak checking of queries)

### CODE REVIEW

File: src/ebonite/repository/metadata/local.py [hunk 1]
```diff
@@ -219,7 +218,7 @@ def get_project_by_id(self, id) -> Project:
     def create_project(self, project: Project) -> Project:
         if self.get_project_by_name(project.name) is not None:
             raise ExistingProjectError(project)
-        project._id = str(uuid.uuid4())
+        project._id = len(self.data.projects)
```
Comment (mike0sv) [inline src/ebonite/repository/metadata/local.py]:
```create_project()
create_project()
create_project()
delete_project_by_id(1)
create_project()
p = get_project_by_id(2)???
```

File: tox.ini [hunk 2]
```diff
@@ -44,9 +44,12 @@ deps =
     coverage==4.5.4
     ; coveralls 5.x is buggy for now
 commands =
-    pytest --cov --cov-report= --cov-append --disable-warnings -m "not docker" tests
+    pytest --cov --cov-report= --cov-append --disable-warnings -m "not docker and not tf_v1" tests
     pytest --cov --cov-report= --cov-append --disable-warnings requirements_tests
     pytest --cov --cov-report= --cov-append --disable-warnings -m docker tests
+    pip uninstall -y tensorflow
+    pip install tensorflow==1.15.2
```
Comment (mike0sv) [inline tox.ini]: maybe <2?

Comment (i-a-andrianov) [reply tox.ini]: fixed

File: src/ebonite/repository/metadata/local.py (hunk 1 above)
Comment (i-a-andrianov) [reply src/ebonite/repository/metadata/local.py]: fixed

File: src/ebonite/repository/metadata/local.py [hunk 3]
```diff
@@ -21,40 +21,55 @@
...
-        for e in environments.values():
+        for e in (environments or {}).values():
             self.add_environment(e)
 
-        for i in instances.values():
+        for i in (instances or {}).values():
             self.add_instance(i)
 
+    def get_and_increment(self, name):
+        next_id = getattr(self, name)
```
Comment (mike0sv) [inline src/ebonite/repository/metadata/local.py]: thread safety?

**Pull request merged**